### DISCUSSION TITLE
Huge memory usage

### DISCUSSION THREAD

**Discussion by zetiti10** [Arduino-IRremote/Arduino-IRremote#1243] [Help] [answered]
Hello!

I'm working on a project based on an Arduino Mega 2560. I need to send and receive IR codes (to control and a device, and to be controlled by a remote control). I only use the library for this:
```cpp
if (m_sensor.decode()) {
  switch (m_sensor.decodedIRData.decodedRawData) {
  case 4244768519:
    m_television.toggle(true);
    break;

  default:
    break;
  }
}
```
and this:
```cpp
m_IRSender.sendNEC(0x44C1, 0x87, 3);
```
My program is available on GitHub: [https://github.com/zetiti10/Domotique-chambre](https://github.com/zetiti10/Domotique-chambre).

Unfortunately, the program memory gets full at startup. I searched the origin and I found… this library.
I'm using PlatformIO in Visual Studio Code, so I can "Inspect" the compiled program. There is a global variable named `irparams` that use 1.5 KB: it's 19% of the 8 KB RAM on the Arduino Mega:
![Memory_inspection](https://github.com/Arduino-IRremote/Arduino-IRremote/assets/72339615/85afab4a-35d2-4138-9332-ea18e342eef1)

With the library:
![With_IRRemote](https://github.com/Arduino-IRremote/Arduino-IRremote/assets/72339615/a0536561-cbdd-43f6-ad24-030e4724da00)
Without the library
![Without_IRRemote](https://github.com/Arduino-IRremote/Arduino-IRremote/assets/72339615/e7d49904-7501-443e-9fc0-334221af4c90)

So my question is: is it possible to use the library without this huge global variable? Or, should I use another library?

**Comment by ArminJo** [comment, marked as answer]
You can specify RAW_BUFFER_LENGTH to change RAM usage.
